operators/tokenizer/bpe_kernels.h

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+#include "ustring.h"
+
+#include <string>
+#include <vector>
+
+struct BpeModelConf {
+  static const char kModel_GPT2[];
+  static const char kModel_Roberta[];
+  static const char kModel_CLIP[];
+
+  const char* name_{kModel_GPT2};
+  const char* unk_token_{"<|endoftext|>"};
+  const char* bos_token_{"<|endoftext|>"};
+  const char* eos_token_{"<|endoftext|>"};
+  const char* pad_token_{nullptr};
+
+  std::string GetSpecialTokens() const;
+};
+
+class BpeModel;
+
+struct KernelBpeTokenizer : BaseKernel {
+  KernelBpeTokenizer(const OrtApi& api, const OrtKernelInfo& info, const BpeModelConf& conf);
+
+  void Compute(const ortc::Tensor<std::string>& input,
+               ortc::Tensor<int64_t>& tokenize_output,
+               std::optional<ortc::Tensor<int64_t>*> attention_mask,
```

Comment:
`optional<Tensor>` or `Tensor*` both imply "optional Tensor", but what does `optional<Tensor*>` mean?

Reply:
std::optional is used to indict this is an optional output in the ONNX operator in LiteCustomOp design. but I don't know why std::optional<...> & is not used. @RandySheriffH

Reply:
put another way, can the `Tensor*` contained within the `optional<Tensor*>` be null? if so, why bother with the `optional`? if not, why is it a pointer?

yeah, maybe we can pass a `Tensor&` instead. `optional<std::reference_wrapper<Tensor>>` could be a possibility.

Reply:
LiteCustomOp needs a type like std::optional to know whether this ONNX operator output is optional. The current 'protocol' is std::optional<Tensor*> so the function prototype here has to align with it. But I agree std::optional<Tensor*> can be improved in LiteCustomOp in the future.